### PR TITLE
Mark what is provisional in the design overview 

### PR DESCRIPTION
**Pull request by josh11b**
Also references were added to clarify that some features are not provisional.

### CODE REVIEW

File: docs/design/README.md [hunk 1]
```diff
@@ -967,6 +997,9 @@ the assignment.
 Unlike C++, these assignments are statements, not expressions, and don't return
 a value.
 
+> **Note:** The semantics of assignment are provisional. See pending proposal
+> [#821: Values, variables, pointers, and references](https://github.com/carbon-language/carbon-lang/pull/821).
+
```
Comment (chandlerc) [inline docs/design/README.md:1002]: Not sure it even makes sense to include #821 vs. just marking as provisional?

**Review by chandlerc — COMMENTED** [review]
Generally looks good to me. Want to give a chance for some others to check that they don't spot missing places here.

File: docs/design/README.md (hunk 1 above)
Comment (josh11b) [reply docs/design/README.md:1002]: It leaves a breadcrumb to see where the content from the README came from, and where to go to read more.

File: docs/design/README.md [hunk 2]
```diff
@@ -2906,6 +2987,9 @@ marker.
 
 ### ABI and dynamic linking
 
+> **Note:** This is provisional, no design for this has been through the
+> proposal process yet.
```
Comment (jonmeow) [inline docs/design/README.md]: I think this isn't provisional, at least for what's written:

- Stable ABI: https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/goals.md#stable-language-and-library-abi
- Interop: https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/interoperability/philosophy_and_goals.md#references, #175

File: docs/design/README.md [hunk 3]
```diff
@@ -2932,6 +3016,9 @@ ABI compatibility.
 
 ### Operator overloading
 
+> **Note:** This is provisional, no design for this has been through the
+> proposal process yet.
```
Comment (jonmeow) [inline docs/design/README.md:3025]: I thought this had been designed out? It's designed through several of the expression proposals:

#1083
#1178 
#845 
#702 
#820

**Review by jonmeow — APPROVED** [review]

File: docs/design/README.md (hunk 2 above)
Comment (jonmeow) [reply docs/design/README.md]: Maybe instead say something like "This reflects goals and plans, as we don't have a specific implementation in place yet."

File: docs/design/README.md (hunk 3 above)
Comment (josh11b) [reply docs/design/README.md:3025]: Carbon operator overloading is designed, but operator overloading interop with C++ isn't, unless I'm mistaken?

Comment (jonmeow) [reply docs/design/README.md:3025]: Ah, sorry, I was missing context on this section.

File: docs/design/README.md (hunk 2 above)
Comment (josh11b) [reply docs/design/README.md]: Is https://github.com/carbon-language/carbon-lang/pull/1984/commits/d66a6410f0c296536dd19254e52efb414b2c7636 make this better?

File: docs/design/README.md [hunk 4]
```diff
@@ -3014,6 +3014,11 @@ available across this interop boundary will of course be restricted to what is
 expressible in the C ABI, and types may need explicit markers to have guaranteed
 ABI compatibility.
 
+> References:
+>
+> -   [Goals: Stable language and library ABI](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/goals.md#stable-language-and-library-abi)
```
Comment (jonmeow) [inline docs/design/README.md]:
```suggestion
> -   [Goals: Stable language and library ABI non-goal](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/goals.md#stable-language-and-library-abi)
```

File: docs/design/README.md [hunk 5]
```diff
@@ -521,6 +533,10 @@ type `Optional(T*)`.
 [stricter pointer provenance](https://www.ralfj.de/blog/2022/04/11/provenance-exposed.html)
 or restrictions on casts between pointers and integers.
 
+> **Note:** While the syntax for pointers has been decided, the semantics of
+> pointers is provisional, as is the syntax for optionals. See pending proposal
```
Comment (geoffromer) [inline docs/design/README.md]: Grammar nit

```suggestion
> pointers are provisional, as is the syntax for optionals. See pending proposal
```

**Pull request merged**